hnn_core/network.py

```diff
@@ -1253,23 +1253,70 @@ def add_connection(self, src_gids, target_gids, loc, receptor,
 
         self.connectivity.append(deepcopy(conn))
 
-    def clear_connectivity(self):
-        """Remove all connections defined in Network.connectivity
+    def _clear_connectivity(self, src_types):
+        """Remove connections with src_type in Network.connectivity.
+
+        Parameters
+        ----------
+        src_types : list
+            Source types of connections to be cleared
+
         """
-        connectivity = list()
-        for conn in self.connectivity:
-            if conn['src_type'] in self.external_drives.keys():
-                connectivity.append(conn)
-        self.connectivity = connectivity
-
-    def clear_drives(self):
-        """Remove all drives defined in Network.connectivity"""
-        connectivity = list()
-        for conn in self.connectivity:
-            if conn['src_type'] not in self.external_drives.keys():
-                connectivity.append(conn)
-        self.external_drives = dict()
-        self.connectivity = connectivity
+        _validate_type(src_types, list, 'src_types', 'list, drives, local')
```

Comment:
Does this need to be updated? I think it should say something like this for the last arg: `'list of drive names or local network connections'`.

Reply:
I think we can remove this check as types are already validated in the public functions - `clear_connectivity()` and `clear_drives()`. Should I remove this check or update it?

Reply:
Good point, go ahead and remove it!

Reply:
Sure will do